pkg/operator/set_operator_split.go

```diff
@@ -0,0 +1,197 @@
+	"github.com/Layr-Labs/eigensdk-go/chainio/clients/elcontracts"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	eigenSdkUtils "github.com/Layr-Labs/eigensdk-go/utils"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/urfave/cli/v2"
+)
+
+func SetOperatorSplitCmd(p utils.Prompter) *cli.Command {
+	var operatorSplitCmd = &cli.Command{
+		Name:  "set-rewards-split",
+		Usage: "Set operator rewards split",
+		Action: func(cCtx *cli.Context) error {
+			return SetOperatorSplit(cCtx, p)
+		},
+		After: telemetry.AfterRunAction(),
+		Flags: getSetOperatorSplitFlags(),
+	}
+
+	return operatorSplitCmd
+}
+
+func SetOperatorSplit(cCtx *cli.Context, p utils.Prompter) error {
+	ctx := cCtx.Context
+	logger := common.GetLogger(cCtx)
+
+	config, err := readAndValidateSetOperatorSplitConfig(cCtx, logger)
+	if err != nil {
+		return eigenSdkUtils.WrapError("failed to read and validate operator split config", err)
+	}
+
+	cCtx.App.Metadata["network"] = config.ChainID.String()
+
+	ethClient, err := ethclient.Dial(config.RPCUrl)
+	if err != nil {
+		return eigenSdkUtils.WrapError("failed to create new eth client", err)
+	}
+
+	if config.Broadcast {
+
+		eLWriter, err := common.GetELWriter(
+			config.OperatorAddress,
+			config.SignerConfig,
+			ethClient,
+			elcontracts.Config{
+				RewardsCoordinatorAddress: config.RewardsCoordinatorAddress,
+			},
+			p,
+			config.ChainID,
+			logger,
+		)
+
+		if err != nil {
+			return eigenSdkUtils.WrapError("failed to get EL writer", err)
+		}
+
+		logger.Infof("Broadcasting set operator transaction...")
+
+		var receipt *types.Receipt
+
+		receipt, err = eLWriter.SetOperatorAVSSplit(ctx, config.OperatorAddress, config.AVSAddress, config.Split, true)
+
+		if err != nil {
+			return eigenSdkUtils.WrapError("failed to process claim", err)
+		}
+
+		logger.Infof("Set operator transaction submitted successfully")
+		common.PrintTransactionInfo(receipt.TxHash.String(), config.ChainID)
+	} else {
+		noSendTxOpts := common.GetNoSendTxOpts(config.OperatorAddress)
+		_, _, contractBindings, err := elcontracts.BuildClients(elcontracts.Config{
+			RewardsCoordinatorAddress: config.RewardsCoordinatorAddress,
+		}, ethClient, nil, logger, nil)
+		if err != nil {
+			return err
+		}
+
+		code, err := ethClient.CodeAt(ctx, config.OperatorAddress, nil)
+		if err != nil {
+			return eigenSdkUtils.WrapError("failed to get code at address", err)
+		}
+		if len(code) > 0 {
+			// Claimer is a smart contract
+			noSendTxOpts.GasLimit = 150_000
+		}
+		var unsignedTx *types.Transaction
+		unsignedTx, err = contractBindings.RewardsCoordinator.SetOperatorAVSSplit(noSendTxOpts, config.OperatorAddress, config.AVSAddress, config.Split)
+
+		if err != nil {
+			return eigenSdkUtils.WrapError("failed to create unsigned tx", err)
+		}
+		if config.OutputType == string(common.OutputType_Calldata) {
```

Comment:
We should print something in the else of it, which will be non broadcast, non call data scenario. Something like this would set the split to the value provided by user.